tests/DataValues/DataValuesTest.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+namespace DataValues\Tests;
+
+use Comparable;
+use DataValues\DataValue;
+use Exception;
+use Hashable;
+use Immutable;
+use PHPUnit\Framework\TestCase;
+use ReflectionClass;
+use Serializable;
+
+/**
+ * Base for unit tests for DataValue implementing classes.
+ *
+ * @since 0.1
+ *
+ * @license GPL-2.0-or-later
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+abstract class DataValuesTest extends TestCase {
```

Comment:
I'm not convoinced we really need to copy all these tests to here, I wonder what @lucaswerkmeister thinks about it...

Reply:
Yeah, I don’t think copying the tests is the best way forward – after all, the DataValues release notes say:

> * Removed `DataValueTest` (create a copy if you need it, though better refactor away the bad design)

Some of the abstract tests here can just be removed as completely pointless, for instance:

```php
$this->assertInstanceOf( $this->getClass(), $dataItem );

$this->assertInstanceOf( Immutable::class, $value );

$this->assertTrue( true );
```

It’s better to put specific, meaningful tests in the particular test classes.